includes/base/controls-stack.php

```diff
@@ -1762,7 +1735,27 @@ public function print_template() {
 	 * @since 1.4.0
 	 * @access protected
 	 */
-	protected function _register_controls() {}
+	protected function _register_controls() {
+		Plugin::$instance->modules_manager->get_modules( 'dev-tools' )->deprecation->deprecated_function( __METHOD__, '2.9.0', __CLASS__ . '::register_controls' );
```

Comment:
```suggestion
		Plugin::$instance->modules_manager->get_modules( 'dev-tools' )->deprecation->deprecated_function( __METHOD__, '3.1.0', __CLASS__ . '::register_controls' );
```

Reply:
Add parentheses after all replacements methods like this: `__CLASS__ . '::register_controls()`